### PR TITLE
Rewriting page creator and html renderer

### LINKED ISSUE #40: Support inline HTML elements in documentation
Both Markdown and Wiki syntaxes support inline HTML elements. We need to define a custom HTML render that supports an appropriate CustomDocTag.

### LINKED ISSUE #51: Rename tabs to something more Scala-like
Probably something like: type members, value members (vals/vars/objects/enums), methods, givens, extensions.

### LINKED ISSUE #157: Change position of source links
Instead of showing it inside tab, we should show source links below class documentation.

### LINKED ISSUE #158: Improve appearance of linear supertypes / known subtypes
See https://scala3doc.s3.eu-central-1.amazonaws.com/pr-master/self/scala3doc/example/level2/-documentation/index.html .

### PR DESCRIPTION
**Pull request by pikinier20**
closes #158
closes #51 
closes #157
closes #40

### CODE REVIEW

**Comment by pikinier20** [comment]
There's only frontend API for rendering inline HTML, we need to handle inline HTML elements in ScalaDocSupport

File: src/main/scala/dotty/dokka/tasty/comments/MarkdownConverter.scala [hunk 1]
```diff
@@ -32,10 +32,11 @@ class MarkdownConverter(val r: Reflection)(owner: r.Symbol) {
...
     case n: mda.Paragraph =>
-      if n.getParent.isInstanceOf[mdu.Document]
-      && !Option(n.getPrevious).exists(_.isInstanceOf[mda.Heading])
-      then
-        emit(dkkd.Br.INSTANCE)
+      // if n.getParent.isInstanceOf[mdu.Document]
+      // && !Option(n.getPrevious).exists(_.isInstanceOf[mda.Heading])
+      // then
+      //   emit(dkkd.Br.INSTANCE)
+      // I've commented it because it inserts unnecessary BRs which are breaking layout
```
Comment (romanowski) [inline src/main/scala/dotty/dokka/tasty/comments/MarkdownConverter.scala]: @abgruszecki Can you take a look?

File: src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala [hunk 2]
```diff
@@ -245,87 +133,264 @@ class ScalaPageCreator(
                     }
                 }
                 case o => bdr.text(s"TODO: $o")
+        }
+
+        def buildBlock = (builder: ScalaPageContentBuilder#ScalaDocumentableContentBuilder) => builder
+            .group(kind = ContentKind.Cover, sourceSets = c.getSourceSets.asScala.toSet) { gbdr => gbdr
+                .cover(c.getName)()
+                .sourceSetDependentHint(Set(c.getDri), c.getSourceSets.asScala.toSet) { sbdr => 
+                    val s1 = sbdr
```
Comment (romanowski) [inline src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala:118]: strange formatting

File: src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala [hunk 3]
```diff
@@ -245,87 +133,264 @@ class ScalaPageCreator(
...
+                    .sourceSetDependentHint(
+                        dri = Set(elem.getDri), 
+                        sourceSets = elem.getSourceSets.asScala.toSet, 
+                        kind = ContentKind.SourceSetDependentHint
+                    ){ srcsetbdr => srcsetbdr
+                        .contentForBrief(elem)
+                        .signature(elem)
+                    }
+                }
+                .groupingBlock(
```
Comment (romanowski) [inline src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala]: Can we split this rendering into few function calls using extension methods?

**Review by romanowski — APPROVED** [review]

File: src/main/scala/dotty/dokka/tasty/comments/MarkdownConverter.scala (hunk 1 above)
Comment (abgruszecki) [reply src/main/scala/dotty/dokka/tasty/comments/MarkdownConverter.scala]: This looks OK, but I'd remove those lines instead. They were added in the first place to separate text paragraphs with an empty line, and they don't seem necessary now. Analogous code in wiki.Converter should be cleaned up as well: https://github.com/lampepfl/scala3doc/blob/4963d716c4f4f5484a8806545ecff9657014ba98/src/main/scala/dotty/dokka/tasty/comments/wiki/Converter.scala#L40-L43 (note that the `topLevel` parameter will, I believe, become useless if this code is removed).

This is a good page to see the current look: https://scala3doc.s3.eu-central-1.amazonaws.com/pr-167/self/scala3doc/tests/-a/index.html . Compare with: https://scala3doc.s3.eu-central-1.amazonaws.com/pr-167/self/scala3doc/tests/-b/index.html , which is rendered from Wiki syntax.

File: project/project/metals.sbt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+// DO NOT EDIT! This file is auto-generated.
+// This file enables sbt-bloop to create bloop config files.
+
+addSbtPlugin("ch.epfl.scala" % "sbt-bloop" % "1.4.3-23-550c6c0a")
```
Comment (abgruszecki) [inline project/project/metals.sbt]: Can we remove this file?

File: project/project/project/metals.sbt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+// DO NOT EDIT! This file is auto-generated.
+// This file enables sbt-bloop to create bloop config files.
+
+addSbtPlugin("ch.epfl.scala" % "sbt-bloop" % "1.4.3-23-550c6c0a")
```
Comment (abgruszecki) [inline project/project/project/metals.sbt]: And can we remove this file as well?

File: src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala [hunk 6]
```diff
@@ -245,87 +133,264 @@ class ScalaPageCreator(
...
-                    .list(subtypes)(contentForType)
-                }
-            }
+            val platforms = d.getSourceSets.asScala.toSet
 
-        def addedContent = ScalaPageContentBuilder(commentsToContentConverter, signatureProvider, logger).contentForDocumentable(
-            clazz, 
-            buildBlock = buildBlock
-        )
+            val description = tags.collect{ case (pd, d: Description) => (pd, d) }.groupBy(_(0)).map( (key, value) => key -> value.map(_(1)))
```
Comment (abgruszecki) [inline src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala]: I'm not 100% sure that this is the correct place, but I guess that the first `Description` tag here should be dropped. The first tag was previously treated as the short description, and wasn't displayed on the class pages. Now it is, which results in pages looking like https://scala3doc.s3.eu-central-1.amazonaws.com/pr-167/self/scala3doc/tests/-b/index.html and https://scala3doc.s3.eu-central-1.amazonaws.com/pr-167/self/scala3doc/tests/-a/index.html (note how paragraphs are duplicated).

**Review by abgruszecki — COMMENTED** [review]
Looks really good! It's really nice that we rewrote the renderer to Scala and fixed a bunch of issues in one swoop. I have, as usual, only minor nitpicks.

File: src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala (hunk 6 above)
Comment (pikinier20) [reply src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala]: Done

File: project/project/project/metals.sbt (hunk 5 above)
Comment (pikinier20) [reply project/project/project/metals.sbt]: Removed

File: project/project/metals.sbt (hunk 4 above)
Comment (pikinier20) [reply project/project/metals.sbt]: Removed

File: src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala (hunk 3 above)
Comment (pikinier20) [reply src/main/scala/dotty/dokka/translators/ScalaPageCreator.scala]: Done

**Pull request merged**